src/bank.rs

```diff
@@ -277,19 +295,66 @@ impl Bank {
 
     /// Process a Transaction. This is used for unit tests and simply calls the vector Bank::process_transactions method.
     pub fn process_transaction(&self, tx: &Transaction) -> Result<()> {
-        match self.process_transactions(&[tx.clone()])[0] {
+        let rv = match self.process_transactions(&[tx.clone()])[0] {
             Err(ref e) => {
                 info!("process_transaction error: {:?}", e);
                 Err((*e).clone())
             }
             Ok(_) => Ok(()),
+        };
+        if rv.is_ok() {
```

Comment:
This is a surprising change to make given its documentation describes this method as "simply calls the vector Bank::process_transactions method".  Is it justified?

Reply:
@garious process_transaction has a side-effect on those locked accounts because of the race condition with poh.  this method is just for unit testing, i think we should `#[cfg(test)]` on top of it